NDTensors/src/default_storage.jl

```diff
@@ -0,0 +1,39 @@
+## This is a fil which specifies the default storage type provided some set of parameters
+## The parameters are the element type and storage type
+function default_storage_type(
+  storetype::Type{<:TensorStorage}=Dense,
+  eltype::Type=Float64,
+  datatype::Type{<:AbstractArray}=Vector,
+  index=Tuple{Int},
+)
+  #TODO check if indices are block sparse or not
+  type = nothing
+  try
+    type = storetype{eltype,datatype{eltype}}
+  catch e
+    println("Cannot construct storetype", storetype)
+    throw(e)
+  end
+  return type
+end
+
+default_storage_type(eltype::Type{<:Number}) = default_storage_type(Dense, eltype)
```

Comment:
I would write this as:
```julia
default_storage_type(eltype::Type{<:Number}) = default_storage_type(default_data_type(eltype))
```
and then define:
```julia
default_data_type(eltype::Type{<:Number}) = Vector{eltype}
```

Reply:
This makes a lot of sense, I have updated that code, thanks!